src/main/java/com/tabnine/statusBar/StatusBarEmptySymbolGenerator.java

```diff
@@ -0,0 +1,20 @@
+package com.tabnine.statusBar;
+
+import java.util.Arrays;
+import java.util.List;
+
+// Workaround to address the issue where IJ does not render the width of the status bar icon
+// correctly when the icon changes
+// because it doesn't detect a change in the widget text (it renders it correctly just after
+// document change occurs).
+// Creating a different empty text each time will solve the problem.
+public class StatusBarEmptySymbolGenerator {
```

Comment:
this is very bote 🤮 
if there's really nothing else that solves it... 🤷🏻‍♂️ 
I'd try to contact jb's support about this anyway

Reply:
great. believe me, I tried anything. nothing solves this.